src/ser/mod.rs

```diff
@@ -520,7 +544,16 @@ impl<'a, W: io::Write> ser::Serializer for &'a mut Serializer<W> {
             self.is_empty = Some(len == 0);
         }
 
-        self.start_indent()?;
+        if let Some((ref config, ref mut pretty)) = self.pretty {
```

Comment:
can this be done more flexible? For example,
```rust
fn start_indent_if(&mut self, condition: impl Fn(&Config) -> bool) {
...
}
fn start_indent(&mut self) {
  self.start_indent_if(|_| true)
}
```
And the same for `fn indent()` and `fn end_indent()`. This would allow us to re-use the code better.

Reply:
Yeah that was exactly what I though but I didn't wanted to make signature change without some people agreeing first.